ethers/signer.nim

```diff
@@ -1,15 +1,16 @@
+import pkg/questionable
 import ./basics
 import ./provider
 
 export basics
 
+{.push raises: [].}
+
 type
   Signer* = ref object of RootObj
     lastSeenNonce: ?UInt256
     populateLock: AsyncLock
-
-type
-  SignerError* = object of EthersError
+  SignerError* = object of ProviderError
```

Comment:
I'd rather not have `SignerError` be a `ProviderError`. Mostly because a signer doesn't have to be associated with a provider. For instance, a wallet can sign stuff without involving a provider.

Reply:
I had another look at this, and separated `ProviderError` from `SignerError` so we can keep their functionality separate, as Ledger does. Some signer methods, eg `getChainId` now can raise both `ProviderError` and `SignerError`. `reverts` additionally checks for `SignerError` now as well 👍